app/src/main/java/mozilla/lockbox/autofill/FillResponseBuilder.kt

```diff
@@ -97,25 +97,26 @@ open class FillResponseBuilder(
         val datasetBuilder = Dataset.Builder()
 
         val title = titleFromHostname(credential.hostname)
+        val username = credential.username ?: context.getString(R.string.no_username)
 
         parsedStructure.usernameId?.let {
             val presentation = RemoteViews(context.packageName, R.layout.autofill_item)
                 .apply {
-                    setTextViewText(R.id.autofillValue, credential.username)
+                    setTextViewText(R.id.autofillValue, username)
                     setTextViewText(R.id.hostname, title)
                 }
 
             datasetBuilder.setValue(
                 it,
-                AutofillValue.forText(credential.username),
+                AutofillValue.forText(credential.username ?: ""),
```

Comment:
what happens when the `credential.username` is null?

Reply:
When the `credential.username` is `null`, the empty string is used instead.

This means that the input field would be filled with the empty string rather than "null". (i.e. `null.toString()`)

Reply:
@eliserichards I've added steps to reproduce in the original issue #488